### ISSUE TITLE
Clarification on ?include assymetry for relationship operations

### ISSUE THREAD

**Issue by cmeeren** [json-api/json-api#1491]
As I read the spec:
* When calling the relationship `related` link `GET /articles/1/comments`, the primary data is `comment`, and if you want the comment authors, you `?include=author`
* When calling the relationship `self` link `GET /articles/1/relationships/comments` (as well as `POST`, `PATCH`, and `DELETE`), the primary data are just resource identifiers, and you have to `include=comments.author` (which will include both the actual comment resources as well as their authors).

In other words, for the relationship `related` link, `?include` works similarly to a resource `self` link, i.e. relative to the primary data of the response, but for a relationship `self` link, any `include` path must necessarily start with the name of the current relationship (e.g. `comments` in the example above).

Is this correct? Is this asymmetry intended?

**Comment by cmeeren** [comment]
> for a relationship `self` link, any `include` path must necessarily start with the name of the current relationship (e.g. `comments` in the example above).

A conflicting but more flexible interpretation is that the "include context" is the "parent resource", and that it would be possible to include other `article` relationships, e.g. `?include=author`. Is this possible? The spec seems silent about it. **Update:** That would conflict with full linkage, so I guess it's not possible.

**Comment by freddrake** [comment]
Does it even make sense to use ``include`` with a relationship's ``self`` link?

If not, that should probably be clarified in the spec so this question is answered for future implementors.

**Comment by cmeeren** [comment]
> Does it even make sense to use `include` with a relationship's `self` link?

It's explicitly supported by the spec (where I got my example):

![image](https://user-images.githubusercontent.com/7766733/106710592-78530900-65f6-11eb-8efe-8be4071b2e39.png)

**Comment by freddrake** [comment]
@cmeeren: Once again it is proven that I can't read!  Thank you for pointing out the specific exposition explaining what I didn't understand correctly.

Regardless of whether the asymmetry you identified is intended, I suspect it can't reasonably be changed because of the full linkage requirement.  There is some language about sparse field sets creating an exception to that, but I'm not convinced I understand it.

**Comment by cmeeren** [comment]
Related: #1492

Sparse fieldsets:
* If you say `GET /articles?include=author`, you get articles as primary data and authors as secondary data. There is full linkage.
* If you say `GET /articles?include=author&fields[article]=title`, you get articles as primary data and authors as secondary data, but since you only show the `title` of each article and not the `author` relationship, there is not full linkage. This is the allowed exception. It would, of course, be generally impossible in this case to know which authors correspond to which articles, but it's allowed.

**Comment by freddrake** [comment]
@cmeeren: Thank you!  Another case where my implementation doesn't get it right.  I've only been including resources where the full linkage is supported by the response document.  Fortunately, this feels very much like an edge case no sane client would choose to deal with.
